### PR TITLE
bpo-44340: Add support for building with clang full/thin lto

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue44340

### CODE REVIEW

**Comment by pablogsal** [comment]
Make sure you use @tiran solution for regenerating autoconf or otherwise is going to cause some incompatibilities (https://bugs.python.org/issue44035)

**Comment by corona10** [comment]
> Make sure you use @tiran solution for regenerating autoconf or otherwise is going to cause some incompatibilities

Thanks, I will check it out

**Comment by corona10** [comment]
@pablogsal  cc @tiran 
Okay, it perfectly works :)

File: configure.ac
Comment (corona10) [inline configure.ac:1378]: If the Py_LTO_POLICY is the default, it will follow the default LTO option of compiler

**Comment by corona10** [comment]
cc @holmanb

File: configure.ac
Comment (corona10) [inline configure.ac:1446]: @holmanb
Should we check the full option also?
IMHO, `-flto` itself can be the alternative to the `full` option.

Comment (holmanb) [reply configure.ac:1446]: I don't have strong opinions on this. Assuming `-flto` when `full` is passed as an option is a sensible assumption, since `full` is the default lto policy on clang, and no other compilers that I'm aware of have multiple lto implementations. 

It may lead to confusion in the future if, for example, clang changes default --flto policy to be thin, or if other compilers produce alternative lto implementations that have different naming schemes.  That said, unless anyone else has stronger opinions, I'd say leave it as is.

Comment (gpshead) [reply configure.ac:1446]: agreed.

**Review by gpshead — APPROVED** [review]

**Comment by gpshead** [comment]
I believe you pushed a bunch of unrelated commits to the branch.

**Comment by corona10** [comment]
> I believe you pushed a bunch of unrelated commits to the branch.

Thanks, I make a lot of people noise ;)
I've rebased the PR.

**Review by corona10 — COMMENTED** [review]
I am now waiting for @pablogsal  's review who is the release manager of Python 3.11 :)

**Review by pablogsal — APPROVED** [review]
Great job @corona10 and @holmanb !

**Pull request merged**
